pkg/iter/entry_iterator.go

```diff
@@ -729,12 +728,21 @@ func (i *reverseEntryIterator) StreamHash() uint64 {
 
 func (i *reverseEntryIterator) Error() error { return nil }
 
-func (i *reverseEntryIterator) Close() error {
+func (i *reverseEntryIterator) release() {
+	if i.buf == nil {
+		return
+	}
+
 	if i.buf.entries != nil {
+		// preserve the underlying slice before releasing to pool
```

Comment:
The comment is a bit weird.  Should we just reslice when we get ?

Reply:
We could do this after `Get`'ing, yeah -- I believe that has equivalent results. The important part is that we don't `nil` it here because the reason we're using a pool is to preserve it. Does that make sense?

Reply:
Yeah I would just re-slice entries are never nil.